File: AIM/AIM/UI/Connectors/TrackListConnector.m
//
//  TrackListConnector.m
//  AIM
//
//  Created by Oskar Szydlowski on 05.03.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

#import "TrackListConnector.h"

#import "TrackListPresenter.h"
#import "TrackListViewController.h"

@interface TrackListConnector () {
    
    TrackListPresenter *presenter;
    TrackListViewController *viewController;
    
}

@end

@implementation TrackListConnector

- (id)initWithUseCaseFactory:(id<UseCaseProducing>)useCaseFactory {
    self = [super init];
    if (self != nil) {
        self.useCaseFactory = useCaseFactory;
    }
    return self;
}

- (UIViewController *)trackListViewController {
    presenter = [[TrackListPresenter new] initWithConnector:self useCaseFactory:self.useCaseFactory];
    viewController = [[TrackListViewController new] initWithPresenter:presenter];
    presenter.view = viewController;
    
    return viewController;
}

@end

File: AIM/AIM/UI/Connectors/TrackListConnector.swift
//
//  TrackListConnector.swift
//  AIM
//
//  Created by Oskar Szydlowski on 28.02.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

import UIKit

class TrackListConnector {
    
    private let useCaseFactory: UseCaseProducing
    
    init(useCaseFactory: UseCaseProducing) {
        self.useCaseFactory = useCaseFactory
    }
    
    func trackListViewController() -> UIViewController {
        let presenter = TrackListPresenter(connector: self, useCaseFactory: useCaseFactory)
        let viewController = TrackListViewController(presenter: presenter)
        presenter.view = viewController
        
        return viewController
    }
}
